include/Authentication.hpp

```diff
@@ -0,0 +1,59 @@
+#ifndef AUTHENTICATION_HPP
+#define AUTHENTICATION_HPP
+
+#include <QObject>
+#include <QString>
+
+class User : public QObject
+{
+    Q_OBJECT
+    Q_PROPERTY(QString username MEMBER username READ getUsername NOTIFY usernameChanged);
+    Q_PROPERTY(QString password MEMBER password READ getPassword NOTIFY passwordChanged);
+    Q_PROPERTY(bool isAuthenticated MEMBER authenticated READ isAuthenticated NOTIFY authenticatedChanged);
+
+public:
+    User(QObject *parent = nullptr);
+    User(const User &other);
+    ~User() {};
+
+    const QString &getUsername() const;
+    const QString &getPassword() const;
+
+    bool isAuthenticated() const;
+
+    User &operator=(const User &other);
+
+    void setUsername(QString &name);
```

Comment:
[F.16: For “in” parameters, pass cheaply-copied types by value and others by reference to const](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f16-for-in-parameters-pass-cheaply-copied-types-by-value-and-others-by-reference-to-const)